src/pieces/logs/serverLog.ts

```diff
@@ -309,6 +310,41 @@ async function processMessageDelete(msg: Message | PartialMessage, serverLog: Te
 	}
 
 	serverLog.send({ embeds: [embed], files: attachments });
+
+	const bot = msg.client;
+	handleExpDetract(bot, msg);
+}
+
+async function handleExpDetract(bot: Client, msg: Message | PartialMessage) {
+	const user: SageUser = await msg.author.client.mongo.collection(DB.USERS).findOne({ discordId: msg.author.id });
+	console.log(user);
+	// bot.mongo.collection(DB.USERS).findOneAndUpdate(
+	// 	{ user },
+	// 	{ $inc: { count: -1, curExp: +1 } }
+	// );
+	if (user.curExp < user.levelExp) {
+		bot.mongo.collection(DB.USERS).findOneAndUpdate(
+			{ discordId: msg.author.id },
+			{ $inc: { count: 0, curExp: +1 } }
+		);
+	} else { // we can't have negative exp
+		bot.mongo.collection(DB.USERS).findOneAndUpdate(
+			{ discordId: msg.author.id },
+			{ $inc: { count: 0, curExp: 0 } }
+		);
+	}
```

Comment:
Why does anything have to happen if the xp would become negative? Unless I misunderstand how to use `{ $inc... }`

Reply:
Initially it was to prevent the percentage being negative when the s;check here command is used. I'll most likely change this so the level rolls back as we discussed earlier